src/main/java/net/mcreator/ui/views/editor/image/ImageMakerView.java

```diff
@@ -207,31 +208,49 @@ public void save() {
 
 	public void saveAs() {
 		Image image = canvasRenderer.render();
-		Object[] options = TextureType.getTypes(false);
-		int n = JOptionPane.showOptionDialog(mcreator, L10N.t("dialog.image_maker.texture_kind"),
-				L10N.t("dialog.image_maker.texture_type"), JOptionPane.YES_NO_CANCEL_OPTION,
-				JOptionPane.QUESTION_MESSAGE, null, options, options[0]);
-		String namec = VOptionPane.showInputDialog(mcreator, L10N.t("dialog.image_maker.enter_name"),
-				L10N.t("dialog.image_maker.image_name"), null, new OptionPaneValidatior() {
-
-					@Override public ValidationResult validate(JComponent component) {
-						return new RegistryNameValidator((VTextField) component,
-								L10N.t("dialog.image_maker.texture_name")).validate();
-					}
-				});
-		if (namec != null && n != -1) {
-			File exportFile = mcreator.getFolderManager()
-					.getTextureFile(RegistryNameFixer.fix(namec), TextureType.getTextureType(n, false));
-
-			if (exportFile.isFile())
-				JOptionPane.showMessageDialog(mcreator, L10N.t("dialog.image_maker.texture_type_name_exists"),
-						L10N.t("dialog.image_maker.resource_error"), JOptionPane.ERROR_MESSAGE);
-			else
-				FileIO.writeImageToPNGFile(ImageUtils.toBufferedImage(image), exportFile);
-			this.image = exportFile;
-			this.name = this.image.getName();
-			refreshTab();
-		}
+

```

Comment:
As you made custom dialog now, this dialog could be both name and type in same prompt, so two dialogs don't open when saving as